nb-adapters/adapter-pulsar/pom.xml

```diff
@@ -34,7 +34,7 @@
     </description>
 
     <properties>
-        <pulsar.version>3.2.1</pulsar.version>
+        <pulsar.version>3.0.7</pulsar.version>
```

Comment:
@yabinmeng could you take a look at this to ensure if its compatible?

Reply:
@msmygit In Pulsar branch 3.0.7 is LTS and is aligned for us. The project is not releasing anything on 3.1, 3.2, and 3.3. Version 4.0 was recently released.

Reply:
It is best to double check with streaming team. I'm not really sure

Reply:
They have been asked. We'll take this change now and can adjust at their direction later.